backend/app/services/search.py

```diff
@@ -68,88 +55,115 @@ async def post(self, query, indice=None, size=10000, start_date=None, end_date=N
                     today = datetime.today().date()
                     seven_days_ago = today - timedelta(days=7)
                     if start_date and start_date > seven_days_ago:
-                        previous_results = []
+                        previous_results = {}
                     else:
                         new_end_date = min(end_date, seven_days_ago) if end_date else seven_days_ago
-                        query['query']['bool']['filter']['range'][timestamp_field]['lte'] = str(new_end_date)
+                        query['query']['bool']['filter'][0]['range'][timestamp_field]['lte'] = str(new_end_date)
                         if start_date:
-                            query['query']['bool']['filter']['range'][timestamp_field]['gte'] = str(start_date)
+                            query['query']['bool']['filter'][0]['range'][timestamp_field]['gte'] = str(start_date)
                         if start_date is None:
                             response = await self.prev_es.search(
                                 index=self.prev_index+"*",
                                 body=jsonable_encoder(query),
                                 size=size)
-                            previous_results = response['hits']['hits']
+                            previous_results = {"data":response['hits']['hits'], "total":response['hits']["total"]["value"]}
                         else:
-                            previous_results = await self.scan_indices(self.prev_es, self.prev_index, query, timestamp_field, start_date, new_end_date, size)
+                            response = await self.prev_es.search(
+                                index=self.prev_index+"*",
+                                body=jsonable_encoder(query),
+                                size=size)
+                            previous_results = {"data":response['hits']['hits'], "total":response['hits']["total"]["value"]}
+                           # previous_results = await self.scan_indices(self.prev_es, self.prev_index, query, timestamp_field, start_date, new_end_date, size, offset)
                 if self.prev_es and self.new_es:
                     self.new_index = self.new_index_prefix + (self.new_index if indice is None else indice)
                     today = datetime.today().date()
                     seven_days_ago = today - timedelta(days=7)
                     if end_date and end_date < seven_days_ago:
-                        new_results = []
+                        new_results = {}
```

Comment:
Why is this being changed to a dictionary?

Reply:
Seems now it has extra values, I am guessing to deal with pagination, because now it includes the `total`